qml/Tutorial/TutorialRight.qml

```diff
@@ -24,7 +24,7 @@ TutorialPage {
     property string usageScenario
 
     // When on phone or tablet, fade out as the drag progresses
-    opacityOverride: usageScenario === "desktop" ? 1 : 1 - stage.rightEdgeDragProgress * 2
+    opacityOverride: stage.spreadShown ? 0 : usageScenario === "desktop" ? 1 : 1 - stage.rightEdgeDragProgress * 2
```

Comment:
This only hides the tutorial, it doesn't skip it. You'll want to find a way to mark the edge as finished instead.

Also, nested ternaries are super ugly. Would not recommend.

Reply:
no it does mark it as finished, look: https://github.com/ubports/unity8/blob/xenial_-_edge/qml/Tutorial/TutorialPage.qml#L49 that calls hide() that then triggers the animation and triggers the onStopped https://github.com/ubports/unity8/blob/xenial_-_edge/qml/Tutorial/TutorialPage.qml#L83

Reply:
Also, nested ternaries does not really matter as the logic would be the same ether way if you use if or whatnot. also: https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340

Reply:
Okay, I see what's going on now. Try the following test:

```
        function test_rightTutorialSkippedBySpread() {
            // Ensures that the right tutorial is skipped by opening the Spread,
            // but no others.
            AccountsService.demoEdgesCompleted = ["left"];
            var tutorialLeft = findChild(shell, "tutorialLeft");
            var tutorialLeftTimer = findInvisibleChild(tutorialLeft, "tutorialLeftTimer");
            tutorialLeftTimer.interval = 99999;
            AccountsService.demoEdgesCompleted = [];
            compare(AccountsService.demoEdgesCompleted, []);
            wait(2000);
            ApplicationManager.startApplication("gallery-app");
            ApplicationManager.startApplication("facebook-webapp");
            ApplicationManager.startApplication("camera-app");
            touchFlick(shell, shell.width, halfHeight, 0, halfHeight);

            compare(AccountsService.demoEdgesCompleted, ["right"])
        }
```

This fails, `demoEdgesCompleted` appears to be `["left","right"]` with the single flick. We're skipping both the left and right tutorial at the same time.

Reply:
but this is unrelated to my changed, seems to be some cache issues or what not. but that's a scenario that wont happen, we will never end up in this case.